annotation-server/src/guidelines/guidelines.service.ts

```diff
@@ -0,0 +1,189 @@
+                        recommendations[row][col].backgroundColor,
+                    );
+                    if (
+                        !implication ||
+                        implication.replace(' ', '').toLowerCase() === 'n/a' ||
+                        !recommendation ||
+                        recommendation.replace(' ', '').toLowerCase() === 'n/a'
+                    ) {
+                        continue;
+                    }
+                    guideline.implication = implication;
+                    guideline.recommendation = recommendation;
+                    guideline.warningLevel = warningLevel;
+                    guideline.genePhenotype = genePhenotype;
+                    guideline.medication = medication;
+                    guidelines.push(guideline);
+                }
+            }
+        }
+
+        this.guidelinesRepository.save(guidelines);
+
+        this.clearMaps();
+    }
+
+    private getWarningLevelFromColor(
+        color?: sheets_v4.Schema$Color,
+    ): WarningLevel | null {
+        if (!color) return null;
+        const [red, green, blue] = [color.red, color.green, color.blue];
+        if (!red && green === 1 && !blue) return WarningLevel.GREEN;
+        if (red === 1 && green === 1 && !blue) return WarningLevel.YELLOW;
+        if (red === 1 && !green && !blue) return WarningLevel.RED;
+        if (red === green && red === blue && blue === green) return null; // any shade of gray or transparent/unset background (undefined)
+        this.logger.warn('Sheet cell has unknown color');
+        return null;
+    }
+
+    private async findMedication(name: string): Promise<Medication> {
+        if (!name) return null;
+        name = name.trim().toLowerCase();
+        if (this.hashedMedications.has(name)) {
+            return this.hashedMedications.get(name);
+        }
+        try {
+            const medication = await this.medicationsService.getOne({
+                where: { name: ILike(name) },
+            });
+
+            this.hashedMedications.set(name, medication);
+            return medication;
+        } catch (error) {
+            // TODO: consider proper error handling
+            this.logger.error(`Medication ${name} not found in Drugbank data.`);
+            this.hashedMedications.set(name, null);
+            return null;
+        }
+    }
+
+    private async findGenePhenotypes(
+        geneSymbolName: string,
+    ): Promise<Array<Set<GenePhenotype>>> {
+        if (!geneSymbolName) return [];
+        geneSymbolName = geneSymbolName.trim().toLowerCase();
+        if (this.hashedGenePhenotypes.has(geneSymbolName)) {
+            return this.hashedGenePhenotypes.get(geneSymbolName);
+        }
+        const geneSymbol = await this.genePhenotypesService.getOne({
+            where: { name: ILike(geneSymbolName) },
+            relations: ['genePhenotypes', 'genePhenotypes.phenotype'],
+        });
+        // TODO: consider proper error handling
```

Comment:
Nit: Create an issue on GitHub and reference the issue number here so that we don't forget about this.

Reply:
We're already working on implementing this #256